### PR TITLE
Fix ngx-datatable doesn't filtering the previous pages bug

### PR DESCRIPTION
**Pull request by Sinan997**
Resolves https://github.com/abpframework/abp/issues/18834

This pr is resolves: https://support.abp.io/QA/Questions/6469

### CODE REVIEW

**Review by masum-ulu — APPROVED** [review]

**Comment by oykuermann** [comment]
This issue is not resolved: 

[29.01.2024 11_11.webm](https://github.com/abpframework/abp/assets/142304401/e07f5575-7986-446d-9ad6-ae5c9cfc58a2)

**Comment by oykuermann** [comment]
This issue occurs on the Audit Logs & Security Logs page: 

[29.01.2024 12_17.webm](https://github.com/abpframework/abp/assets/142304401/839181d1-582f-4b2c-8276-afec28182a6e)
image

**Review by oykuermann — CHANGES_REQUESTED** [review]
I'll check after the changes

**Comment by oykuermann** [comment]
The issue ( https://github.com/abpframework/abp/pull/18871#issuecomment-1914278500 ) will be resolved with https://github.com/abpframework/abp/issues/18877

**Review by oykuermann — APPROVED** [review]

**Review by masum-ulu — APPROVED** [review]

File: npm/ng-packs/packages/core/src/lib/services/list.service.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class ListService<QueryParamsType = ABP.PageQueryParams | any> implements
     this._query$.next({
       filter: this._filter || undefined,
       maxResultCount: this._maxResultCount,
-      skipCount: this._page * this._maxResultCount,
+      skipCount: this._filter ? 0 : this._page * this._maxResultCount,
```
Comment (mustafapsd) [inline npm/ng-packs/packages/core/src/lib/services/list.service.ts:153]: This line prevents the user from changing the page when the `filter` is not null. 

Case:

1. Search for something with the search input.
2. Try to change the page.
3. The page will be changed, but `skipCount` will be zero:
Screenshot 2024-03-02 at 19 54 10

4. When you remove the search parameter, `skipCount` will be correct.
Screenshot 2024-03-02 at 19 57 03

**Review by mustafapsd — COMMENTED** [review]
This PR causing a bug.
@oykuermann 
@masumulu28

File: npm/ng-packs/packages/core/src/lib/services/list.service.ts (hunk 1 above)
Comment (Sinan997) [reply npm/ng-packs/packages/core/src/lib/services/list.service.ts:153]: Hi Mustafa, i have created an issue for this problem. https://github.com/abpframework/abp/issues/19194

**Pull request merged**